xgi/classes/simplicialcomplex.py

```diff
@@ -0,0 +1,332 @@
+                except:
+                    raise XGIError("The simplex cannot be cast to a frozenset.")
+
+                self._edge_attr[uid].update(attr)
+                self._edge_attr[uid].update(dd)
+
+                # add subfaces 
+                faces = self._subfaces(simplex)
+                self.add_simplices_from(faces)
+
+    def add_weighted_simplices_from(self, ebunch_to_add, weight="weight", **attr):
+        """Add weighted simplices in `ebunch_to_add` with specified weight attr
+
+        Parameters
+        ----------
+        ebunch_to_add : container of simplices
+            Each simplex given in the list or container will be added
+            to the graph. The simplices must be given as containers.
+        weight : string, optional (default= 'weight')
+            The attribute name for the simplex weights to be added.
+        attr : keyword arguments, optional (default= no attributes)
+            simplex attributes to add/update for all simplices.
+
+        See Also
+        --------
+        add_simplex : add a single simplex
+        add_simplices_from : add multiple simplices
+
+        Notes
+        -----
+        Adding the same simplex twice will add it only once.
+        """
+
+        try:
+            self.add_edges_from(
+                ((edge[:-1], {weight: edge[-1]}) for edge in ebunch_to_add), **attr
+            )
+        except:
+            XGIError("Empty or invalid edges specified.")
+
+    def remove_simplex_id(self, id):
+        """Remove a simplex with a given id.
+
+        This also removes all simplices of which this simplex is face,
+        to preserve the simplicial complex structure.
+
+        Parameters
+        ----------
+        id : Hashable
+            edge ID to remove
+
+        Raises
+        ------
+        XGIError
+            If no edge has that ID.
+
+        See Also
+        --------
+        remove_edges_from : remove a collection of edges
+        """
+        try:
+
+            # remove all simplices that contain simplex
+            supfaces_ids = self._supfaces_id(self._edge[id])
+            self.remove_simplex_ids_from(supfaces_ids)
+
+            # remove simplex
+            for node in self.edges.members(id):
+                self._node[node].remove(id)
+            del self._edge[id]
+            del self._edge_attr[id]
+
+        except KeyError as e:
+            raise XGIError(f"Simplex {id} is not in the Simplicialcomplex") from e
+
+    def remove_simplex_ids_from(self, ebunch):
+        """Remove all simplicies specified in ebunch.
+
+        Parameters
+        ----------
+        ebunch: list or container of hashables
+            Each edge id given in the list or container will be removed
+            from the hypergraph.
+
+        See Also
+        --------
+        remove_simplex_id : remove a single simplex by ID
+
+        Notes
+        -----
+        Will fail silently if an edge in ebunch is not in the simplicial complex.
```

Comment:
It seems inconsistent to me that `remove_simplex_ids_from` fails silently but `remove_simplex_id` raises an exception. I think the two of them should have the same behavior. Perhaps both raise a warning (not exception)?

Reply:
Fair point. I actually copied the behaviour we had for those in the Hypergraph class. I think Nicholas' idea of not raising a exception when removing a bunch that you get from a dataset, and there might be many that are not in the Hypergraph, but raising many warnings/exceptions might be a pain. But when you remove a single one I guess you're expected to know which one it is. I don't have a strong opinion. What do you think?

Reply:
I see the problem. Then I think this solution makes sense. Writing it down explicitly (as it is at the moment in the note) seems fair.

Reply:
Let's leave it for now then!